src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -83,69 +104,76 @@ struct LocationStructureWrapper : public chip::app::Clusters::ServiceArea::Struc
+    {
+        mapID = aMapId;
+    }
+
+    /**
+     * @brief setter for location name
      * @param[in] aLocationName A human readable name for this location (empty string if not used).
-     * @param[in] aFloorNumber The floor level of this location - use negative values for below ground.
-     * @param[in] aAreaTypeTag A common namespace Area tag - indicates an association of the location with an indoor or outdoor area
-     * of a home.
-     * @param[in] aLandmarkTag A common namespace Landmark tag - indicates an association of the location with a home landmark.
-     * @param[in] aPositionTag A common namespace Position tag - indicates the position of the location with respect to the
-     * landmark.
-     * @param[in] aSurfaceTag A common namespace Floor Surface tag - indicates an association of the location with a surface type.
-     *
-     * @note Requirements regarding what combinations of fields and values are valid are not checked by this class.
+     * @param[in] doInfoStructCleanup If true, locationInfo.locationInfo will be set to null if
+     *            aLocationName is an empty string and floorNumber and areaType are null.
+     *            This generates the 'canonical' form specified by the cluster requirements.
      * @note If aLocationName is larger than kLocationNameMaxSize, it will be truncated.
-     * @note If aLocationName is an empty string and aFloorNumber and aAreaTypeTag are null, locationInfo will be set to null.
      */
-    void Set(uint32_t aLocationId, const DataModel::Nullable<uint8_t> & aMapId, const CharSpan & aLocationName,
-             const DataModel::Nullable<int16_t> & aFloorNumber, const DataModel::Nullable<AreaTypeTag> & aAreaType,
-             const DataModel::Nullable<LandmarkTag> & aLandmarkTag, const DataModel::Nullable<PositionTag> & aPositionTag,
-             const DataModel::Nullable<FloorSurfaceTag> & aSurfaceTag)
+    void SetLocationName(const CharSpan & aLocationName, bool doInfoStructCleanup)
     {
-        locationID = aLocationId;
-        mapID      = aMapId;
-
-        // If there is at least one non-null value for locationInfo, add it to the location structure.
-        if ((!aLocationName.empty()) || (!aFloorNumber.IsNull()) || (!aAreaType.IsNull()))
+        // does the locationInfo structure need to be added?
+        if ((!aLocationName.empty()) && 
+            (locationInfo.locationInfo.IsNull()))
         {
-            // Create a home location info structure and fill it in except for the location name. This is done below.
+            // Create a location info structure
             locationInfo.locationInfo.SetNonNull(Structs::LocationDescriptorStruct::Type());
-
-            locationInfo.locationInfo.Value().floorNumber = aFloorNumber;
-            locationInfo.locationInfo.Value().areaType    = aAreaType;
         }
-        else
+
+        // does the locationInfo structure need to be removed and set to NULL?
+        if ((doInfoStructCleanup) &&
```

Comment:
Why the parentheses around a bareword boolean?

In general, these if conditions all seem way over-parenthesized.

Reply:
Yes, my coding style is to use lot's of parenthesis in logic statements.  At best it avoids errors due to the compiler executing in the wrong order, it clarifies the intent, and yes, it does clutter things up a bit.